### ISSUE TITLE
Unintuitive search results

### ISSUE THREAD

**Issue by ZibingZhang** [sandboxnu/searchneu-v1#177]
Described here: https://github.com/sandboxnu/searchneu/issues/152, closed but not fixed

The first result when searching `cs 2510` is `CS 2511: Lab for CS 2510`. I would expect the search results for `cs2510` and `cs 2510` to be the same. Probably just needs a regex and some normalization in the BE.

**Comment by hankewyczz** [comment]
tldr: we know about this, but won't change it back in the short term as it's a workaround for a fatal performance bug in one of our dependencies

Longer story: the way we used to handle these queries (ie. queries which are a subject + course code) was by normalizing the input and delegating to our database. 

However, our ORM has a performance bug - if its connection pool (to the DB) ever becomes full, it will not recover until the app is fully restarted. As a result, we've removed the database calls which are directly dependent on user input (if a lot of users query for single courses, it will likely flood the ORM, leading to this bug). 

So, for the time being, we're delegating all searches to Elasticsearch. When the bug is fixed, we'll update the ORM and switch back.

**Comment by ZibingZhang** [comment]
Do you think it'd be possible to do the normalization in the FE for the time being?

I would assume that most people search by the `XXXX ####` pattern

**Comment by hankewyczz** [comment]
@ZibingZhang feel free to use the staging environment: https://searchneu.vercel.app until the change is approved and deployed (note that the staging environment may be overwritten with other changes, so it may lose this functionality in the future)

- A course subject/code combo which exists will return only that course
  - https://searchneu.vercel.app/NEU/202330/search/cs2500
- A subject/code combo in which the subject is valid but the code doesn't exist will return nothing
  - https://searchneu.vercel.app/NEU/202330/search/cs2502
- A combo where the subject doesn't exist will default to a normal search
  - https://searchneu.vercel.app/NEU/202330/search/aa2500
